init/init.c

```diff
@@ -469,26 +473,43 @@ int load_module(struct kmod_ctx *ctx, const char *module_path) {
     return 0;
 }
 
+// is_kernel_module is a helper function to check if a path has a .ko or .ko.xz 
+// extension indicating that it is a kernel module file. Files ending in .ko.xz 
+// are compressed kernel modules. 
+bool is_kernel_module(const char *fpath) {
```

Comment:
This function assumes kmod_ext_length <= kmod_xz_ext_length.
Suggest a check_extension function instead.

Reply:
I wanted to avoid something like a "check_extension" function specifically so we could avoid getting the string length of the paths so many times as a performance optimization. I also felt it was safe to assume `kmod_ext_length < kmod_xz_ext_length` since we are the ones that define `kmod_ext` and `kmod_xz_ext` earlier in the file and I don't see those extensions ever changing. Thoughts?